actors/builtin/market/market_test.go

```diff
@@ -350,3 +439,135 @@ func (h *marketActorTestHarness) expectProviderControlAddressesAndValidateCaller
+
+	params := market.WithdrawBalanceParams{
+		ProviderOrClientAddress: provider,
+		Amount:                  big.Add(withDrawAmt, big.NewInt(1)),
+	}
+
+	rt.ExpectSend(owner, builtin.MethodSend, nil, expectedSend, nil, exitcode.Ok)
+	rt.Call(h.WithdrawBalance, &params)
+	rt.Verify()
+}
+
+func(h *marketActorTestHarness) withdrawClientBalanceOK(rt *mock.Runtime, client address.Address, withDrawAmt, expectedSend abi.TokenAmount) {
+	rt.SetCaller(client, builtin.AccountActorCodeID)
+	rt.ExpectValidateCallerType(builtin.CallerTypesSignable...)
+	rt.ExpectSend(client, builtin.MethodSend, nil, expectedSend, nil, exitcode.Ok)
+
+	params := market.WithdrawBalanceParams{
+		ProviderOrClientAddress: client,
+		Amount:                  withDrawAmt,
+	}
+
+	rt.Call(h.WithdrawBalance, &params)
+	rt.Verify()
+}
+
+func (h *marketActorTestHarness) publishDealOK(rt *mock.Runtime, deal *market.DealProposal, owner, worker, provider address.Address) abi.DealID {
+	rt.SetCaller(worker, builtin.AccountActorCodeID)
+	rt.ExpectValidateCallerType(builtin.CallerTypesSignable...)
+	rt.ExpectSend(
+		provider,
+		builtin.MethodsMiner.ControlAddresses,
+		nil,
+		big.Zero(),
+		&miner.GetControlAddressesReturn{Owner: owner, Worker: worker},
+		exitcode.Ok,
+	)
+
+	//  create a client proposal with a valid signature
+	buf := bytes.Buffer{}
+	require.NoError(h.t, deal.MarshalCBOR(&buf), "failed to marshal deal proposal")
+	sig := acrypto.Signature{Type: acrypto.SigTypeBLS, Data: []byte("does not matter")}
+	clientProposal := market.ClientDealProposal{*deal, sig}
+	params := &market.PublishStorageDealsParams{[]market.ClientDealProposal{clientProposal}}
+
+	// expect a call to verify the above signature
+	rt.ExpectVerifySignature(sig, deal.Client, buf.Bytes(), nil)
+
+	ret := rt.Call(h.PublishStorageDeals, params)
+	rt.Verify()
+
+	resp, ok := ret.(*market.PublishStorageDealsReturn)
+	require.True(h.t, ok, "unexpected type returned from call to PublishStorageDeals")
+	require.Len(h.t, resp.IDs, 1)
+
+	dealId := resp.IDs[0]
+	require.NotNil(h.t, h.mustGetDealProposal(rt, dealId))
+
+	return dealId
+}
+
+func (h *marketActorTestHarness) generateUnVerifiedDealProposal(client, provider address.Address, startEpoch, endEpoch abi.ChainEpoch) *market.DealProposal{
+	buf := make([]byte, binary.MaxVarintLen64)
+	binary.PutVarint(buf, int64(rand.Int()))
+	hash, err := mh.Sum(buf, mh.SHA2_256, -1)
+	require.NoError(h.t, err)
+
+	pieceCid := cid.NewCidV0(hash)
+	pieceSize := abi.PaddedPieceSize(2048)
+	storagePerEpoch := big.NewInt(int64(rand.Intn(1000) + 1))
+	clientCollateral := big.NewInt(int64(rand.Intn(1000) + 1))
+	providerCollateral := big.NewInt(int64(rand.Intn(1000) + 1))
+
+	return &market.DealProposal{pieceCid, pieceSize, false,client, provider, startEpoch,
+		endEpoch, storagePerEpoch, providerCollateral, clientCollateral}
+}
+
+func (h *marketActorTestHarness) activeDealOK(rt *mock.Runtime, dealID abi.DealID, sectorExpiry abi.ChainEpoch, minerAddr address.Address) {
```

Comment:
`activateDeals`. Similarly, make it plural now.

Reply:
This is done.